devwilliamy/coverland
# Adds preorder for seat cover products

https://www.figma.com/design/f0PiJaMbj3moDGXq45iAA1/Coverland-Website-Design?node-id=5335-58941&t=dze4U1PKBE9Kduyp-0

Testing Instructions:
https://coverland-2-0-git-john-preview-seat-c-d004ee-coverland-3615f02b.vercel.app/
:
-check various seat covers products
-try to find weird behavior/bugs with preorder notices/display (especially with dates)
-add preorder items (seat covers only) to cart
-checkout preorder items
--* pay attention for any odd numbers (cart totals, order summary, delivery date, etc.)
-complete your order using various payments (Stripe, Paypal, etc)
-check your Order Confirmation email: 
-- *Order id should include "PRE" if it includes a preorder item ie Order #CL-TEST-PRE-240710-SE-0215
-- *expected delivery date, preorder item totals/discounts, total discount, total

Further Testing:
- Make sure you are able to add car covers too (My code shouldn't have affected car cover products whatsoever)
- Try adding odd combinations of items (Note: weird behavior here

Core changes:
- seat cover page:
when a color is clicked (seat cover product) it will check for preorder status

- in stock:
![Screenshot 2024-07-10 191558](https://github.com/devwilliamy/coverland/assets/171850605/e7cf7d9a-1831-4cec-a0c6-26a284a27d88)

-preorder:
![Screenshot 2024-07-10 191612](https://github.com/devwilliamy/coverland/assets/171850605/ce00b9ef-ae75-4f38-bfe7-a20593bfb8fe)

Notes:
-please check for case when preorder is false, but preorder_date and preorder_discount are null:
-please check for case when preorder is true, but preorder_date and preorder_discount are null:
-found issue when preorder_date is null:
![image](https://github.com/devwilliamy/coverland/assets/171850605/92cf0248-e766-4dab-8411-9feb94336a62)


when add to cart is clicked normal behavior should occur.

when "Preorder and save big is clicked" should populate to the figma preorder cart slider:
![image](https://github.com/devwilliamy/coverland/assets/171850605/253f5a8b-2c28-4649-9fca-547b6d8eb923)

Note: the "5 weeks" should be based on preorder_date.
Again, when preorder_date is null:
![image](https://github.com/devwilliamy/coverland/assets/171850605/14a790e8-c93f-4b50-b402-d0327b8252a1)

Please look at additional changes in comments below. I don't want to overload this section so I will continue in comments..

TODO:
-modify order detail page to reflect preorder / preorder discount status
-make changes to shipping email so that it also gets modified data (order totals, discounts, delivery date, etc)
